### PR TITLE
Installation issues

### PR DESCRIPTION
**Pull request by zhtk**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,6 +20,30 @@ Dependencies
...
+* https://github.com/Juniper/contrail-python-api
+	```sh
+	git clone https://github.com/Juniper/contrail-python-api
+	cd contrail-python-api
+	sudo python setup.py install
+	```
+
+* `vnc_api` and `cfgm_common`. They are not available as standalone packages, so the best idea is to search for
+	`api-lib/dist/vnc_api-0.1.dev0.tar.gz` and `config/common/dist/cfgm_common-0.1.dev0.tar.gz` in
+	your Contrail build directory and then copy these files to OpenStack server. Then issue
```
Comment (renkam) [inline README.md:35]: Please mention here that those files are located on Contrail node after successful Contrail build.

File: README.md [hunk 2]
```diff
@@ -62,7 +83,9 @@ _Note:_ `CONTRAIL_SRC_DIR` _is_ `/opt/stack/contrail` _when using_ **Contrail-in
 	* In file `$CONTRAIL_SRC_DIR/contrail-web-core/config/config.global.js` replace entry `config.identityManager.ip` to point to **OpenStack node**
 	* In file `$CONTRAIL_SRC_DIR/contrail-web-core/config/config.global.js` replace entry `config.featurePkg.webController.path` to `'$CONTRAIL_SRC_DIR/contrail-web-controller';`
 4. Verify that Contrail is working correctly
-5. Install [contrail-python-api](https://github.com/Juniper/contrail-python-api)
+5. Install listed dependencies in OpenStack node
+6. Copy `neutron` directory from the repository to `$OpenStack/neutron`, so the plugin is in `$OpenStack/neutron/neutron/plugins/ml2/drivers/contrail_driver.py`
```
Comment (renkam) [inline README.md:87]: Please use uppercase for "$OpenStack" to be consistent with lines above.

File: devstack-install.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
+# Installation guide - devstack scenario
```
Comment (renkam) [inline devstack-install.md:1]: This looks to me like instructions on how to install OpenStack using devstack so please rename this section accordingly.

File: devstack-install.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+	CONTRAIL_DRIVER_PORT=8082
+	NEUTRON_ENTRY_POINTS='/opt/stack/neutron/neutron.egg-info/entry_points.txt'
+	
+	# Obviously, You should include this:
+	enable_plugin contrail-mechanism-driver https://github.com/zhtk/ContrailMechanismDriver.git
+	```
+
+3. Run `./stack.sh`
+
+## Steps on contrail node
```
Comment (renkam) [inline devstack-install.md:31]: This is covered in README.md file

File: devstack/settings [hunk 5]
```diff
@@ -0,0 +1,7 @@
+# Contrail controller address
+CONTRAIL_DRIVER_CONTROLLER=${CONTRAIL_DRIVER_CONTROLLER:-'10.10.1.247'}
```
Comment (renkam) [inline devstack/settings:2]: Default Contrail address should be more generic (like 127.0.0.1)

File: install_functions.sh [hunk 6]
```diff
@@ -46,9 +46,10 @@ install_plugin()
 {
 	[ -z "$1" ] && { echo "Missing argument for \`install_plugin' function: missing OpenStack install directory"; exit 2; }
 	OpenStack="$1"
-	{ date; echo "Copying plugin (whole dir: $MainDir/neutron) into $OpenStack"; ls -al "$MainDir/neutron"; } >> /tmp/congl.log; 
-	
+	{ date; echo "Copying plugin (whole dir: $MainDir/neutron) into $OpenStack"; ls -al "$MainDir/neutron"; } >> /tmp/congl.log;
+
 	sudo cp -dR "$MainDir/neutron" "$OpenStack"/
+	sudo cp -dR "$MainDir/neutron" "$OpenStack"/neutron
```
Comment (renkam) [inline install_functions.sh:52]: Why copy same content into separate directories?
Are both target locations are correct?

File: devstack/plugin.sh [hunk 7]
```diff
@@ -20,8 +20,22 @@ setup_stack()
 
 		"post-config")
 			source "$NEUTRON_DIR/devstack/plugin.sh"
-			source "$NEUTRON_DIR/devstack/lib/ml2" contrail_driver
-			enable_ml2_extension_driver
+
+			# Insert extension driver
```
Comment (renkam) [inline devstack/plugin.sh:24]: This looks to me very similar as configure_plugin() function from install_functions.sh file. Please avoid code duplication.

**Review by renkam — CHANGES_REQUESTED** [review]

File: install_functions.sh (hunk 6 above)
Comment (zhtk) [reply install_functions.sh:52]: No, probably second one is correct

**Pull request closed without merging**